src/Service.Tests/SqlTests/GraphQLMutationTests/DwSqlGraphQLMutationTests.cs

```diff
@@ -434,6 +434,154 @@ public override async Task DeleteMutationWithOnlyTypename()
         }
+
+        /// <summary>
+        /// <code>Do: </code>insert a new book and return all the books with same publisher
+        /// <code>Check: </code>the intended book is inserted.
+        /// </summary>
+        [TestMethod]
+        public async Task TestStoredProcedureMutationNonEmptyResponse()
+        {
+            string dbQuery = @"
+SELECT CASE
+         WHEN EXISTS (
+           SELECT 1
+           FROM [books] AS [table0]
+           JOIN (
+             SELECT id
+             FROM [publishers]
+             WHERE name = 'Big Company'
+           ) AS [table1] ON [table0].publisher_id = [table1].id
+         ) THEN 'true' ELSE 'false'
+       END AS IsInserted
+            ";
+
+            string graphQLMutationName = "executeInsertAndDisplayAllBooksUnderGivenPublisher";
+            string graphQLMutation = @"
+                mutation{
+                    executeInsertAndDisplayAllBooksUnderGivenPublisher(book_id: 1, title: ""Orange Tomato"" publisher_name: ""Big Company""){
+                        id
+                        title
+                    }
+                }
+            ";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true);
+            string insertedValue = "{\"id\":1,\"title\":\"Orange Tomato\"}";
+            string expected = await GetDatabaseResultAsync(dbQuery);
+            Assert.IsTrue(actual.ToString().Contains(insertedValue));
+            Assert.IsTrue(expected.ToString().Equals("True"));
+        }
+
+        /// <summary>
+        /// <code>Do: </code>insert new Book and return nothing
+        /// <code>Check: </code>if the intended book is inserted in books table
+        /// </summary>
+        [TestMethod]
+        public async Task TestStoredProcedureMutationForInsertion()
+        {
+            string msSqlQuery = @"
+SELECT CASE 
+         WHEN EXISTS (
+           SELECT 1
+           FROM [books] AS [table0]
+           WHERE [table0].[title] = 'Random Book'
+             AND [table0].[publisher_id] = 1234
+         ) THEN 'true' 
+         ELSE 'false'
+       END AS BookExists
+            ";
+
+            string graphQLMutationName = "executeInsertBook";
+            string graphQLMutation = @"
+                mutation {
+                    executeInsertBook(book_id: 10, title: ""Random Book"", publisher_id: 1234 ) {
+                        result
+                    }
+                }
+            ";
+
+            string currentDbResponse = await GetDatabaseResultAsync(msSqlQuery);
+            Assert.AreEqual(currentDbResponse, "False", "Entry should not exist in the db before insertion");
+            JsonElement graphQLResponse = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, isAuthenticated: true);
+
+            // Stored Procedure didn't return anything
+            SqlTestHelper.PerformTestEqualJsonStrings("[]", graphQLResponse.ToString());
```

Comment:
Shouldn't `result` be a field in the response and contain records changed?

Reply:
For stored procedures with no result set returned we return an empty array. This is the relevant code in schema converter that is similar for mssql as well: 
<img width="596" alt="image" src="https://github.com/Azure/data-api-builder/assets/124841904/2aa0f99c-d1c4-48b9-8a42-abaad8504faa">

Reply:
I ran this in debugger and looks like DAB's raw response is the following:

```json
{"data":{"executeInsertBook":[]}}
```

there isn't a "result" field in the response.

Reply:
Discussed offline. This is part of existing implementation of how the result of a query without results is processed which this pr does not address.